### PR TITLE
Feature/48 get latest tweets api

### PR DESCRIPTION
**Pull request by TheoStanica**
Solve [Issue 48](https://github.com/TheoStanica/mark-manager/issues/48)

### CODE REVIEW

File: social/src/routes/twitter/home-timeline.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import express, { Request, Response } from 'express';
+import {
+  BadRequestError,
+  FailedConnectionError,
+  ForbiddenError,
+  requireAuth,
+} from '@tcosmin/common';
```
Comment (RaduAlbastroiu) [inline social/src/routes/twitter/home-timeline.ts:7]: How did you end up choosing this library? Looks like it is a less used one, usually I would suggest look for the most used ones.

File: social/src/routes/twitter/home-timeline.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import twit from 'twit';
+import { UserController } from '../../controllers/user-controller';
+
+const router = express.Router();
+const consumerKey = process.env.TWITTER_CONSUMER_KEY!;
+const consumerSecret = process.env.TWITTER_CONSUMER_SECRET!;
```
Comment (RaduAlbastroiu) [inline social/src/routes/twitter/home-timeline.ts:13]: I believe you have a .env file in your local repository, I would suggest create a new file in the root path of your project called .env.example where you have all the keys without their values. Reason being that everyone who sees you repo will know what keys are necessary in order to run the project.

File: social/src/routes/twitter/home-timeline.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        consumer_key: consumerKey,
+        consumer_secret: consumerSecret,
+        access_token: oauthAccessToken,
+        access_token_secret: oauthAccessTokenSecret,
+      });
+
+      try {
+        const timeline = await T.get('statuses/home_timeline');
+        if (timeline) {
+          res.send(timeline.data);
```
Comment (RaduAlbastroiu) [inline social/src/routes/twitter/home-timeline.ts:35]: In case there are no posts received should you send at least an empty array [] ?

**Review by RaduAlbastroiu — CHANGES_REQUESTED** [review]

File: social/src/routes/twitter/home-timeline.ts (hunk 1 above)
Comment (TheoStanica) [reply social/src/routes/twitter/home-timeline.ts:7]: This is my own library. I use this to share Errors, Events and some Middlewares between microservices

File: social/src/routes/twitter/home-timeline.ts (hunk 2 above)
Comment (TheoStanica) [reply social/src/routes/twitter/home-timeline.ts:13]: Env variables are set inside every microservice config file (X-depl.yaml). These variables are set inside the Kubernetes cluster and microservices will throw an error at startup if one variable is not set

File: social/src/routes/twitter/home-timeline.ts (hunk 3 above)
Comment (TheoStanica) [reply social/src/routes/twitter/home-timeline.ts:35]: fixed

File: social/src/routes/twitter/home-timeline.ts (hunk 2 above)
Comment (TheoStanica) [reply social/src/routes/twitter/home-timeline.ts:13]: Maybe I can list these variables as well as how to set one inside the Kubernetes cluster inside a "How to run the app" section in README.md at a later date

**Comment by TheoStanica** [comment]
Fixed issue

File: social/src/routes/twitter/home-timeline.ts (hunk 2 above)
Comment (RaduAlbastroiu) [reply social/src/routes/twitter/home-timeline.ts:13]: Would be a good thing to have a README for the entire project.

File: social/src/routes/twitter/home-timeline.ts (hunk 1 above)
Comment (RaduAlbastroiu) [reply social/src/routes/twitter/home-timeline.ts:7]: Well done improving the community with your own libraries!

**Review by RaduAlbastroiu — APPROVED** [review]

**Pull request merged**